zoo/birealnet.py

```diff
@@ -0,0 +1,212 @@
+                                    "magnitude_aware_sign",
+                                    kernel_constraint="weight_clip", ))
+    model.add(tf.keras.layers.BatchNormalization())
+
+  @staticmethod
+  def add_larq_connection_block(model, filters_num):
+    """
+    Same method of add_qkeras_connection_block but for a larq network
+    """
+    model.add(
+      lq.layers.QuantConv2D(filters_num, (3, 3), strides=(2, 2), use_bias=False,
+                            padding="same",
+                            input_quantizer="approx_sign",
+                            kernel_quantizer="magnitude_aware_sign",
+                            kernel_constraint="weight_clip"))
+
+    shape_in = model.output_shape[1] * model.output_shape[2] * \
+               model.output_shape[3]
+    shape_out = (model.output_shape[1], model.output_shape[2],
+                 model.output_shape[3] // 2)
+    # Prepare shapes for reshape layers
+    model.add(tf.keras.layers.Flatten())
+    model.add(tf.keras.layers.Reshape(target_shape=(shape_in, 1)))
+    model.add(tf.keras.layers.AvgPool1D(1, strides=2, padding="same"))
+    model.add(tf.keras.layers.Reshape(target_shape=shape_out))
+    model.add(tf.keras.layers.Conv2D(filters_num, (1, 1), padding="same",
+                                     use_bias=False))
+    model.add(tf.keras.layers.BatchNormalization())
+    model.add(tf.keras.layers.BatchNormalization())
+
+  def build(self):
+    """
+    Build the model
+    :return: qkeras and larq models
+    """
+    qkeras_network = self.build_qkeras_birealnet()
+    print("\nQKeras network successfully created")
+    larq_network = self.build_larq_birealnet()
+    print("Larq network successfully created")
+    return qkeras_network, larq_network
+
+  def build_qkeras_birealnet(self):
+    """
+    Build the qkeras version of the birealnet
+    :return: qkeras model of the birealnet
+    """
+    qkeras_biRealNet = tf.keras.models.Sequential()
+    qkeras_biRealNet.add(tf.keras.layers.InputLayer(input_shape=(224, 224, 3)))
+    qkeras_biRealNet.add(
+      tf.keras.layers.Conv2D(64, (7, 7), strides=2, padding="same",
+                             use_bias=False))
+    qkeras_biRealNet.add(tf.keras.layers.BatchNormalization(momentum=0.8))
+    qkeras_biRealNet.add(
+      tf.keras.layers.MaxPool2D((3, 3), strides=2, padding="same"))
+    for _ in range(0, 4):
+      self.add_qkeras_residual_block(qkeras_biRealNet, 64)
+    self.add_qkeras_connection_block(qkeras_biRealNet, 128)
+
+    for _ in range(0, 3):
+      self.add_qkeras_residual_block(qkeras_biRealNet, 128)
+    self.add_qkeras_connection_block(qkeras_biRealNet, 256)
+
+    for _ in range(0, 3):
+      self.add_qkeras_residual_block(qkeras_biRealNet, 256)
+    self.add_qkeras_connection_block(qkeras_biRealNet, 512)
+
+    for _ in range(0, 3):
+      self.add_qkeras_residual_block(qkeras_biRealNet, 512)
+    qkeras_biRealNet.add(tf.keras.layers.AveragePooling2D(pool_size=(7, 7)))
```

Comment:
Any reason of not using power of two numbers for pool_size? For example, (2,2) or (4,4)

We need to be careful here regarding the precision, if the pool_size is not power of two (49=7x7 here). 

```avg = sum(pool_element) / pool_size = sum(pool_element) / 49```, which introduces the precision loss if we use fixed point or integer to hold avg, right.